src/core.rs

```diff
@@ -36,11 +34,83 @@ pub(crate) async fn reply_disconnected(id: &RequestId, stdout_sink: &mut StdoutS
     Ok(())
 }
 
+pub(crate) async fn connect(app_state: &AppState) -> Result<SseClientType> {
+    // this function should try sending a POST request to the sse_url and see if
+    // the server responds with 405 method not supported. If so, it should call
+    // connect_with_sse, otherwise it should call connect_with_streamable.
+    let result = reqwest::Client::new()
+        .post(app_state.url.clone())
+        .header("Accept", "application/json,text/event-stream")
+        .header("Content-Type", "application/json")
+        .body(r#"{"jsonrpc":"2.0","id":"init","method":"initialize","params":{"protocolVersion":"2025-03-26","capabilities":{},"clientInfo":{"name":"test","version":"0.1.0"}}}"#)
+        .send()
+        .await?;
+
+    if result.status() == reqwest::StatusCode::METHOD_NOT_ALLOWED {
+        debug!("Server responded with 405, using SSE transport");
+        return connect_with_sse(app_state).await;
+    } else if result.status().is_success() {
+        debug!("Server responded successfully, using streamable transport");
+        return connect_with_streamable(app_state).await;
+    } else {
+        error!("Server returned unexpected status: {}", result.status());
+        anyhow::bail!("Server returned unexpected status: {}", result.status());
+    }
```

Comment:
we may want to just fallback to sse in that case, since some servers may not respond with 405 🤔

Reply:
I am fine with being conservative for now. All Tidewave servers return 405 already.